R/binary_classification.R

```diff
@@ -80,56 +80,58 @@ logLoss <- function(actual, predicted) {
 
 
 #' Precision
-#' 
+#'
 #' \code{precision} computes proportion of observations predicted to be in the
 #'                  positive class (i.e. the element in \code{predicted} equals 1)
-#'                  that actually belong to the positive class (i.e. the element 
+#'                  that actually belong to the positive class (i.e. the element
 #'                  in \code{actual} equals 1)
-#'     
+#'
 #' @inheritParams params_binary
 #' @export
-#' @seealso \code{\link{recall}} \code{\link{fbeta_score}}
-#' @examples 
+#' @seealso \code{\link{recall}} \code{\link{fbeta_score}} \code{\link{ppv}}
+#' @examples
 #' actual <- c(1, 1, 1, 0, 0, 0)
 #' predicted <- c(1, 1, 1, 1, 1, 1)
 #' precision(actual, predicted)
 precision <- function(actual, predicted) {
-    return(mean(actual[predicted == 1]))
+    cm <- confusion_matrix(actual, predicted)
+    cm$tp / (cm$tp + cm$fp)
```

Comment:
This returns `NaN` if `tp` and `fp` both are 0. This is absolutely fine, but should be clearly documented though. Also note that other toolkits decided to return either 0 or 1 in such a case. There is a discussion about it on SO: https://stats.stackexchange.com/questions/1773/what-are-correct-values-for-precision-and-recall-in-edge-cases.